api/v1alpha1/groupversion_info.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1alpha1 contains API Schema definitions for the provider v1alpha1 API group
+//+kubebuilder:object:generate=true
+//+groupName=provider.giantswarm.io
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// GroupVersion is group version used to register these objects
+	GroupVersion = schema.GroupVersion{Group: "provider.giantswarm.io", Version: "v1alpha1"}
```

Comment:
Not sure about the group. If we wanted this to be an upstream project, we could also consider `x-k8s.io`, but that might be jumping the gun.

Reply:
It might be nice not to have giantswarm specific stuff if we want it to be used by other people, not sure if that's a goal either though. Do you have reasons in mind why it might be hard to switch to `x-k8s.io`?

Reply:
We can't do CRD conversion webhooks if we change the group. Instead it would be a one-way migration. We have plenty of time to decide though.

Reply:
After discussing in SIG Operator ([here](https://gigantic.slack.com/archives/C2MS2VB37/p1624378390074800)), I decided to change the group to `infrastructure.cluster.x-k8s.io`.